File: python-course/com/example/selenium/spree/LoginPage.py
from BasePage import BasePage
from NewsFeedPage import NewsFeedPage


class LoginPage(BasePage):
    def __init__(self, test, navigate_to):
        super(LoginPage, self).__init__(test=test,
                                        url="https://cloudsso.cisco.com/sp/startSSO.ping?SpSessionAuthnAdapterId=standardnomfa&TargetResource=https://sso.cisco.com/autho/login/loginaction.html",
                                        title="Cisco.com Login Page",
                                        navigate_to=navigate_to)

    def assert_on_page(self):
        inner_title = self.test.web_driver.find_element_by_xpath("//*[@id=\"fw-title-nav-login\"]/h1")
        self.test.assertEquals(inner_title.text, "Log in to your account")

    def get_login_button(self):
        return self.test.web_driver.find_element_by_id("login-button")

    def get_user_input(self):
        return self.test.web_driver.find_element_by_xpath("//*[@id=\"userInput\"]")

    def get_password_input(self):
        return self.test.web_driver.find_element_by_id("passwordInput")

    def sign_in(self, username, password):
        if not self.login(username, password):
            return None

        return NewsFeedPage(self.test, False)

    def login(self, username, password):
        self.get_user_input().send_keys(username)
        self.get_login_button().click()

        password_input = self.get_password_input()
        self.test.assertTrue(password_input.is_displayed())
        self.test.assertTrue(password_input.is_enabled())
        password_input.send_keys(password)

        self.get_login_button().click()
        msg = self.test.web_driver.find_element_by_css_selector("#warning-msg > h3")
        return not msg.text == "That login didn't work:"

File: java-course/src/test/java/com/example/selenium/spree/LoginPage.java
package com.example.selenium.spree;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.remote.RemoteWebDriver;
import org.testng.Assert;

public class LoginPage extends BasePage {
    public LoginPage(RemoteWebDriver webDriver, boolean navigateTo) {
        super(webDriver,
                "https://cloudsso.cisco.com/sp/startSSO.ping?SpSessionAuthnAdapterId=standardnomfa&TargetResource=https://sso.cisco.com/autho/login/loginaction.html",
                "Cisco.com Login Page",
                navigateTo);
    }

    @Override
    protected void assertOnPage() {
        WebElement innerTitle = webDriver.findElement(By.xpath("//*[@id=\"fw-title-nav-login\"]/h1"));
        Assert.assertEquals(innerTitle.getText(), "Log in to your account");
    }

    public NewsFeedPage signIn(String username, String password) {
        if (!login(username, password)) {
            return null;
        }
        return new NewsFeedPage();
    }

    public void navigateTo() {
        webDriver.navigate().to(URL);
        Assert.assertEquals(webDriver.getTitle(), "Cisco.com Login Page");
    }

    private WebElement getLoginButton() {
        return webDriver.findElement(By.id("login-button"));
    }

    private WebElement getUserInput() {
        return webDriver.findElement(By.xpath("//*[@id=\"userInput\"]"));
    }

    private WebElement getPasswordInput() {
        return webDriver.findElement(By.id("passwordInput"));
    }

    private boolean login(String username, String password) {
        getUserInput().sendKeys(username);
        getLoginButton().click();

        WebElement passwordInput = getPasswordInput();
        if (!passwordInput.isDisplayed()) {
            return false;
        }

        passwordInput.sendKeys(password);
        getLoginButton().click();
        WebElement element = webDriver.findElement(By.cssSelector("#warning-msg > h3"));
        return !element.getText().equals("That login didn't work:");
    }
}
